src/main/java/organice/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,53 @@
+package organice.logic.parser;
+
+import static organice.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static organice.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+import organice.logic.commands.ListCommand;
+import organice.logic.parser.exceptions.ParseException;
+import organice.model.person.Type;
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Returns the {@code Type} of person in the given {@code ArgumentMultimap}
+     * @throws ParseException if the type is not specified correctly in the input arguments
+     */
+    private static Type parseType(ArgumentMultimap argumentMultimap) throws ParseException {
+        try {
+            return ParserUtil.parseType(argumentMultimap.getValue(PREFIX_TYPE).get());
+        } catch (NoSuchElementException ex) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE));
```

Comment:
since we only call `parseType` when `isTypePresent` returns true, i think can remove the `try-catch` block and just `return ...`  cos there is no case where there is no type specified in this case

Reply:
Actually the `try-catch` block is needed so it catches a `NoSuchElementException` and returns a `ParseException`.

The `NoSuchElementException` occurs when the argumentMultimap cannot retrieve the `Type` from the user input. This exception is not very specific to our application. In the context of our application, the reason why the `Type` cannot be retrieved is because of invalid user inputs resulting in errors while parsing. Hence, a `ParseException` should be thrown here. 

I hope this explanation is clear. If there are any issues do let me know.

Reply:
i thought `NoSuchElementException` occurs when the user dont specify `t/` in the command? Am i correct?

Reply:
it will be thrown when the user dont specify `t/` even if the user specify `t/gibberish`, it will still return true and wont throw `NoSuchElementException` i believe

Reply:
I think I get your point now. What happens when the user specifies an invalid value such as `t/gibberish`?
How will the exception be thrown?

Reply:
Okay noted. Then I think you can just stick to your suggestion.

Reply:
Okayy, cos i was doing testcases, then that catch block is never reached cos impossible?

Reply:
Removed `try...catch` block.